### PR TITLE
chore: setup fastlane

### PR DESCRIPTION
**Pull request by AsCress**
Configures _fastlane_ in our repository to automate uploads to the PlayStore Open Testing track.

## Changes 
- Whenever a new commit is pushed, the release APK is pushed to the _beta_ (open testing) track in the Google PlayStore.

## Screenshots / Recordings  
N/A

**Checklist**: 
- [x] **No hard coding**: I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard coding any value.
- [x] **No end of file edits**: No modifications done at end of resource files `strings.xml`, `dimens.xml` or `colors.xml`.
- [x] **Code reformatting**: I have reformatted code and fixed indentation in every file included in this pull request.
- [x] **No extra space**: My code does not contain any extra lines or extra spaces than the ones that are necessary.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,220 @@
+GEM
```
Comment (adityastic) [inline Gemfile.lock]: We can remove the lockfile. We want to anyway use the latest version of fastlane

File: fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+default_platform(:android)
+platform :android do
+  lane :uploadToOpenTesting do
+    upload_to_play_store(track: "beta",aab:"build/app/outputs/bundle/release/app-release.aab")
```
Comment (adityastic) [inline fastlane/Fastfile]:
```suggestion
    upload_to_play_store(track: "beta",aab:"app/build/outputs/bundle/release/app-release.aab")
```

**Review by adityastic — CHANGES_REQUESTED** [review]

File: fastlane/Appfile [hunk 3]
```diff
@@ -0,0 +1,2 @@
+json_key_file("./android/fastlane.json") # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
```
Comment (adityastic) [inline fastlane/Appfile]:
```suggestion
json_key_file("./scripts/fastlane.json") # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
```

**Review by adityastic — CHANGES_REQUESTED** [review]

File: .github/workflows/push-event.yml [hunk 4]
```diff
@@ -28,6 +28,23 @@ jobs:
...
+          ruby-version: '3.3'
+          bundler-cache: true
+
+      - name: Prepare Bundler
+        if: ${{ github.repository == 'fossasia/pslab-android' }}
+        run: |
+          bundle config path vendor/bundle
+          bundle install --jobs 4 --retry 3
+
+      - name: Pre Checks
```
Comment (adityastic) [inline .github/workflows/push-event.yml]: This might not work for pslab. Maybe remove it for now and we can add it later. I can explain what this does

**Review by adityastic — CHANGES_REQUESTED** [review]

**Comment by AsCress** [comment]
@adityastic Thanks for the review ! I've made the required changes.

**Review by adityastic — APPROVED** [review]

**Pull request merged**